### PR TITLE
EDSC-4295: Adds layerPeriod to gibs tag. Uses layerPeriod to determine time param for layers

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

When we load GIBS imagery, we should only include the full datetime if the product has a period of `Subdaily`. Most products are `Daily`.

### What is the Solution?

Add the `layerPeriod` value to our GIBS tagging job. 
Use the `layerPeriod` value from the GIBS tag to determine if the full datetime should be used, or only the date.

I also updated leaflet, and included the `detectRetina` prop to TileLayer, to improve the visual rendering of our map on higher pixel density screens.

### What areas of the application does this impact?

Granule imagery

# Testing

Open your browser dev tools, switch to the network tab, and filter requests by images. 

On `main` branch, load this URL, http://localhost:8080/search/granules?p=C2208779826-LANCEMODIS&pg[0][v]=f&pg[0][gsk]=-start_date&q=C2208779826-LANCEMODIS&tl=1730131646!3!!

You will see many granule GIBS images that are identical.
Screenshot 2024-10-28 at 12 09 26 PM

Then switch to the `EDSC-4295` branch and refresh the page. You should not see any duplicate images. 
Screenshot 2024-10-28 at 12 10 35 PM

Testing with a collection with a subdaily period is difficult because we rely on tags on production collections. But you can invert the `===` to `!==` in GranuleGridLayerExtended.jsx on line 334 and view this URL (http://localhost:8080/search/granules?p=C2930725014-LARC_CLOUD&pg[0][v]=f&pg[0][gsk]=-start_date&qt=2024-10-13T22%3A21%3A56.279Z%2C2024-10-17T16%3A27%3A26.802Z&fi=TEMPO&as[instrument][0]=TEMPO&tl=1729701652!3!!&lat=36.403453175476805&long=-151.40495681762695), and you will see that the time is included in the image requests

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/Map/GranuleGridLayerExtended.jsx [hunk 1]
```diff
@@ -39,7 +39,8 @@ const config = {
   // eslint-disable-next-line max-len
   gibsUrl: 'https://gibs.earthdata.nasa.gov/wmts/{lprojection}/best/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}',
   // eslint-disable-next-line max-len
-  gibsGranuleUrl: 'http://uat.gibs.earthdata.nasa.gov/wmts/{lprojection}/std/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}'
+  gibsGranuleUrl: 'http://uat.gibs.earthdata.nasa.gov/wmts/{lprojection}/std/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}',
+  debug: true
```
Comment (macrouch) [inline static/src/js/components/Map/GranuleGridLayerExtended.jsx:43]: Remove

**Review by macrouch — COMMENTED** [review]
Change `period` to `layerPeriod`

**Comment by macrouch** [comment]
> Change `period` to `layerPeriod`

This is done

**Review by daniel-zamora — APPROVED** [review]

**Review by trevorlang — APPROVED** [review]

File: static/src/js/components/Map/GranuleGridLayerExtended.jsx [hunk 2]
```diff
@@ -39,7 +39,8 @@ const config = {
   // eslint-disable-next-line max-len
   gibsUrl: 'https://gibs.earthdata.nasa.gov/wmts/{lprojection}/best/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}',
   // eslint-disable-next-line max-len
-  gibsGranuleUrl: 'http://uat.gibs.earthdata.nasa.gov/wmts/{lprojection}/std/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}'
+  gibsGranuleUrl: 'http://uat.gibs.earthdata.nasa.gov/wmts/{lprojection}/std/{product}/default/{time}/{resolution}/{z}/{y}/{x}.{format}',
```
[the remark this answers is not in the document]
Comment (macrouch) [reply static/src/js/components/Map/GranuleGridLayerExtended.jsx]: I don't know why this is in here, but looking at the requests we make from the browser we aren't using the UAT urls.

**Pull request merged**